File: ts/1664/1539.ts
function findKthPositive(arr: number[], k: number): number {
  if (arr[0] > k) return k
  let l = 0, r = arr.length - 1, m = 0
  while (l <= r) {
    m = l + ((r - l) >> 1)
    if (arr[m] - m - 1 >= k) r = m - 1
    else l = m + 1
  }
  return k + l // arr[l - 1] + k - (arr[l - 1] - (l - 1)  - 1)
};

File: py3/1664/1539.py
class Solution:
  def findKthPositive(self, arr: List[int], k: int) -> int:
    if arr[0] > k:
      return k
    l, r, m = 0, len(arr) - 1, 0
    cnt = 0
    while l <= r:
      m = l + (r - l) // 2
      if arr[m] - m - 1 >= k:
        r = m - 1
      else:
        l = m + 1
    return k + l